Dockerfile

```diff
@@ -1,15 +1,45 @@
+# LAYER builder — — — — — — — — — — — — — — — — — — — — — — —
 FROM golang:1.12-alpine AS builder
 ADD go.mod go.sum /app/
 WORKDIR /app/
 RUN apk --no-cache add git upx
 RUN go mod download
 ADD . /app/
 RUN CGO_ENABLED=0 GOOS=linux GOARCH=amd64 go build -ldflags "-X main.version=$(git describe --tags --abbrev=0)" -a -o /trivy cmd/trivy/main.go
```

Comment:
It would be great if `GOARCH` was based on the actual build architecture/a build variable (e.g. this builds fine for a RPI if set to `arm`). 

However, it may be better to solve this later, in a following PR.

Reply:
If you are using docker buildx to build multi-arch the `$TARGETARCH` argument (auto injected but needs to be defined) contains the build arch (`amd64`, `arm64` etc). So basically:

```
ARG TARGETARCH
RUN CGO_ENABLED=0 GOOS=linux GOARCH=${TARGETARCH} go build -ldflags ...
```

should work just fine.

Reply:
agree

> However, it may be better to solve this later, in a following PR.